src/main/java/seedu/address/logic/parser/AddNoteCommandParser.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_IC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddNoteCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.IdentityCardNumber;
+import seedu.address.model.person.IdentityCardNumberMatchesPredicate;
+import seedu.address.model.person.Note;
+
+/**
+ * Parses input arguments and creates a new {@code AddNoteCommand} object
+ */
+public class AddNoteCommandParser implements Parser<AddNoteCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code AddNoteCommand}
+     * and returns a {@code AddNoteCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddNoteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_IC, PREFIX_NOTE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_IC, PREFIX_NOTE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddNoteCommand.MESSAGE_USAGE));
+        }
+
+        IdentityCardNumber ic;
+        try {
+            String note = argMultimap.getValue(PREFIX_NOTE).orElse("");
+
+            boolean isReplace;
+            if (note.contains("-replace")) {
```

Comment:
"-replace" looks like a Magic variable. We should probably make this a constant if we want to do this. Personally this feels like it should belong in the CliSyntax class.

I'd also be more comfortable if we could use `ArgumentTokenizer` to detect for the presence of these flags that don't take values, though I'm not sure if that's supported yet.

Reply:
I think I will just shift it to the CliSyntax class because I am not really sure on how to incorporate -replace into the ArgumentTokenizer class.